core/state/statedb.go

```diff
@@ -251,6 +251,7 @@ func (s *StateDB) StopPrefetcher() {
 	s.prefetcherLock.Lock()
 	if s.prefetcher != nil {
 		s.prefetcher.close()
+		s.prefetcher = nil
```

Comment:
I can't remember why this was removed, and I can not sure why add it back could solve the problem.
Maybe there could be another bug?
We can merge it first and dig deep when we get time.

Reply:
`parlia` is fine.
this `ut` used `ethash` as consensus engine and do `StateIntermediateRoot` on the same `statedb` more than once, would lead the chance to recalculate stateRoot with empty `pendingObjects`

Reply:
Well, IMO, it does make sense to mimic the "reset" behavior as it is done in the `StartPrefetcher` method when `prefetcher` isn't `nil`.

Reply:
yes, reset it to nil is ok to me.